scripts/Muon/GUI/Common/list_selector/list_selector_presenter.py

```diff
@@ -0,0 +1,86 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+
+
+class ListSelectorPresenter(object):
+    def __init__(self, view, model):
+        self.view = view
+        self.model = model
+        self.filter_string = ''
+        self.filter_type = 'Include'
+        self.filter_list = []
+
+        self.view.set_filter_line_edit_changed_action(self.handle_filter_changed)
+        self.view.set_item_selection_changed_action(self.handle_selection_changed)
+        self.view.set_filter_type_combo_changed_action(self.set_filter_type)
+        self.view.set_select_all_checkbox_action(self.handle_select_all_checkbox_changed)
+        self.view.set_row_moved_checkbox_action(self.handle_row_moved)
+
+    def update_view_from_model(self):
+        filtered_list = self.get_filtered_list()
+
+        self.view.clearItems()
+        self.view.addItems(filtered_list)
+
+    def handle_filter_changed(self, filter_string):
+        self.filter_string = filter_string
+        self.update_view_from_model()
+
+    def handle_selection_changed(self, name, state):
+        self.model[name][1] = state
```

Comment:
what is the significance of the 1?

Reply:
The information about each row is stored as a list, the first item is position, the second is whether it is selected and the third is whether it is enabled.

Reply:
could we have variable names at the top of the file and then use them instead?